sampler_options.go

```diff
@@ -0,0 +1,48 @@
+package jaeger
+
+// RemotelyControlledSamplerOption is a function that sets some option on the sampler
+type RemotelyControlledSamplerOption func(sampler *RemotelyControlledSampler)
+
+// RemotelyControlledSamplerOptions is a factory for all available RemotelyControlledSamplerOption's
+var RemotelyControlledSamplerOptions remotelyControlledSamplerOptions
+
+type remotelyControlledSamplerOptions struct{}
+
+// Metrics creates a RemotelyControlledSamplerOption that initializes Metrics on the sampler,
+// which is used to emit statistics.
+func (remotelyControlledSamplerOptions) Metrics(m *Metrics) RemotelyControlledSamplerOption {
```

Comment:
nit: this still allows changing options post-construction, e.g.

```go
s := NewRemotelyControlledSampler(...)
RemotelyControlledSamplerOptions.HostPort("...")(s)
```

A cleaner approach is to move the fields like `metrics`, `maxOperations`, etc. into `remotelyControlledSamplerOptions` and then embed that struct into the sampler struct

The constructor method then becomes
```go
func NewRemotelyControlledSampler(..., options...) {
    opts := remotelyControlledSamplerOptions{}
    for _, o := range options {
        o(&opts)
    }
    // apply defaults
    if opts.metrics == nil {
        opts.metics = NewMetrics() //NB I think we should change it to a pointer rather than a value
    }
    s := &Sampler {
        remotelyControlledSamplerOptions: opts,
    }
}
```

And one last thing. The safest way to apply defaults is **after** the functional options have been executed.  Right now you can screw up by passing `Options.MaxOperations(0)` (unless we want 0 to mean do not cache, in which case this is ok but same principle still applies to other options).